### PR TITLE
Add Client::plot_config + relevant types + example

### PR DESCRIPTION
**Pull request by Dinnerbone**
The example produces this trace with custom plot types:
image

### CODE REVIEW

File: tracy-client/src/plot.rs [hunk 1]
```diff
@@ -6,6 +6,61 @@ use crate::Client;
...
+    Stepped,
+
+    /// Lines will be smooth (interpolating a line between two values).
+    #[default]
+    Smooth,
+}
+
+/// Configuration for how a plot appears in the Tracy profiling UI.
+#[derive(Clone, PartialEq, Debug)]
+pub struct PlotConfiguration {
```
Comment (Ralith) [inline tracy-client/src/plot.rs:39]: Should this be `#[non_exhaustive]`, or maybe even a builder struct with private fields, for forwards compatibility?

Comment (Dinnerbone) [reply tracy-client/src/plot.rs:39]: Builder sounds good! I'll make it so.

Comment (Dinnerbone) [reply tracy-client/src/plot.rs:41]: How's that?

Comment (Ralith) [reply tracy-client/src/plot.rs:41]: LGTM, thanks!

File: tracy-client/src/plot.rs [hunk 2]
```diff
@@ -6,6 +6,89 @@ use crate::Client;
...
+pub enum PlotLineStyle {
+    /// Lines will be stepped (ie look like a staircase).
+    Stepped,
+
+    /// Lines will be smooth (interpolating a line between two values).
+    #[default]
+    Smooth,
+}
+
+/// Configuration for how a plot appears in the Tracy profiling UI.
```
Comment (nagisa) [inline tracy-client/src/plot.rs:39]: Worth adding a code example here. I found it to be pretty common for people to struggle to figure out how to construct stuff, when the construction is not a `fn new()` (even if it is as simple as using `default()` instead.) A plain example would help with that.

**Review by nagisa — APPROVED** [review]

**Comment by nagisa** [comment]
This is all in 0.17.5.

**Pull request merged**